src/Route.ts

```diff
@@ -402,14 +417,48 @@ export default class Route implements IMountableItem {
       );
   }
 
+  private getRequestFingerprint(
+    path: string,
+    variables: Record<string, unknown>,
+    headers: Record<string, unknown>,
+  ) {
+    const hash = createHash('sha1');
+
+    hash.update(path);
+
+    Object.entries(variables).forEach((k, v) => hash.update(`${k}-${v}`));
+    Object.entries(headers).forEach((k, v) => hash.update(`${k}-${v}`));
```

Comment:
Do we want to hash headers?

Reply:
I think there would be a lot of use cases, where hashing headers would be required (e.g. `Authorization` or `Cookie`). Keeping this would make sense to me. Alternatively we could provide a list of headers to include as configuration, but this would require extra configuration from the user. So I think like this it makes the most sense.

Reply:
Good call outs.

I am adding a configuration option to allow users to specify which headers to hash. 

If you don't use `passThroughHeaders` there won't be any headers available to hash anyway, but if your use case has authorization you will be able to do something like this:

```
new GraphQLRestRouter(..., ..., { passThroughHeaders: ['Authorization'], cacheableHeaders: ['Authorization'] });
```

This will specify to include `Authorization` in the request's hash. Configuration name is still TBD.